File: 2363-merge-similar-items/2363.py
class Solution:
    def mergeSimilarItems(self, items1: List[List[int]], items2: List[List[int]]) -> List[List[int]]:
        ans = []
        i1 = sorted(items1)
        i2 = sorted(items2)

        ptr1 = 0
        ptr2 = 0

        while ptr1 < len(i1) and ptr2 < len(i2):
            if i1[ptr1][0] == i2[ptr2][0]:
                ans.append([i1[ptr1][0], i1[ptr1][1]+i2[ptr2][1]])
                ptr1 += 1
                ptr2 += 1
            elif i1[ptr1][0] < i2[ptr2][0]:
                ans.append([i1[ptr1][0], i1[ptr1][1]])
                ptr1 += 1
            else:
                ans.append([i2[ptr2][0], i2[ptr2][1]])
                ptr2 += 1

        while ptr1 < len(i1):
            ans.append(i1[ptr1])
            ptr1 += 1

        while ptr2 < len(i2):
            ans.append(i2[ptr2])
            ptr2 += 1

        return ans

File: 2363-merge-similar-items/2363.cpp
class Solution {
public:
    vector<vector<int>> mergeSimilarItems(vector<vector<int>>& items1, vector<vector<int>>& items2) {
        vector<int> m(1001);
        for (const vector<int>& item : items1) {
            m[item[0]] += item[1];
        }
        for (const vector<int>& item : items2) {
            m[item[0]] += item[1];
        }
        vector<vector<int>> ans;
        for (int i = 1; i < m.size(); ++i) {
            if (m[i] != 0)
                ans.push_back({i, m[i]});
        }
        return ans;
    }
};
